File: android/tutorials/tutorial1/src/main/java/com/ubercab/presidio_training1/root/RootRouter.java
package com.ubercab.presidio_training1.root;

import com.uber.rib.core.ViewRouter;

/** Adds and removes children of {@link RootBuilder.RootScope}. */
public class RootRouter extends ViewRouter<RootView, RootInteractor, RootBuilder.Component> {

  RootRouter(RootView view, RootInteractor interactor, RootBuilder.Component component) {
    super(view, interactor, component);
  }
}

File: android/demos/coroutines/src/main/kotlin/com/uber/rib/root/RootRouter.kt
package com.uber.rib.root

import android.view.View

import com.uber.rib.core.ViewRouter
import com.uber.rib.root.details.DetailsBuilder
import com.uber.rib.root.details.DetailsRouter

/**
 * Adds and removes children of {@link RootBuilder.RootScope}.
 *
 * TODO describe the possible child configurations of this scope.
 */
class RootRouter(
        view: RootView,
        interactor: RootInteractor,
        component: RootBuilder.Component,
        val detailsBuilder: DetailsBuilder,
) : ViewRouter<RootView, RootInteractor>(view, interactor, component) {

    var detailsRouter: DetailsRouter? = null

    fun attachDetails() {
        detailsRouter = detailsBuilder.build(getView())
        attachChild(detailsBuilder)
        view.addView(loggedOutRouter.view)
    }
}
